pkg/controllers/raycluster_controller.go

```diff
@@ -619,3 +626,47 @@ func (r *RayClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	return controller.Complete(r)
 }
+
+func RCCUniqueName(s string) string {
+	return s + "-" + seededHash(controllerName, s)
+}
+
+func deleteDeprecatedObjects(ctx context.Context, r *RayClusterReconciler, cluster *rayv1.RayCluster) error {
```

Comment:
For existing Ray clusters, is my understanding correct the existing resources will be deleted once the operator is upgraded, but these existing Ray clusters will still point to those deleted resources?

Reply:
Ah, yeah. I'll go this route. It should be pretty straight forward